### PR TITLE
Clustering in ML

### PR DESCRIPTION
**Pull request by sepehr-amini-afshar**
Probably gonna add some codes to show results of different clustering methods on the same task.

### CODE REVIEW

**Comment by sarakh1999** [comment]
Hi, 
Thank you very much, your notebook is pretty well!
As a suggestion, you can add some pieces of code implementing different types of gradient methods and compare the results and the consumed time. If it is not challenging for you, implement an easy deep task on an image dataset (such as MNIST) to visualize and compare the produced results better.
You can also write more about different usages. For example, explain the usages in deep neural networks and backpropagation in more detail. 
Thanks again!
